packages/shader-lab/src/ast-node/AstNode.ts

```
@@ -0,0 +1,1001 @@
+
+    const propList = [...(this.content.properties ?? []), ...(this.content.indexes ?? [])]
+      .sort((a, b) => AstNodeUtils.astSortAsc(a.position, b.position))
+      .map((item) => item.serialize(context))
+      .join("");
+
+    if (propName) {
+      if (objName === context.varyingStructInfo.objectName) {
+        const ref = context.varyingStructInfo.reference?.find(
+          (ref) => ref.property.content.variableNode.content.variable === propName
+        );
+        ref && (ref.referenced = true);
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)